cabal-install/Distribution/Client/ProjectConfig/Legacy.hs

```diff
@@ -217,9 +217,9 @@ convertLegacyGlobalConfig
       savedHaddockFlags      = haddockFlags
     } =
     mempty {
-      projectConfigShared        = configAllPackages,
-      projectConfigLocalPackages = configLocalPackages,
-      projectConfigBuildOnly     = configBuildOnly
+      projectConfigBuildOnly   = configBuildOnly,
```

Comment:
(I reordered the fields to match declaration order in `Distribution.Client.ProjectConfig.Types`)